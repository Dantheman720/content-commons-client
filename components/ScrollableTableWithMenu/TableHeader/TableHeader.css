.items_table .ui.sortable.table thead tr th {
  position: sticky;
  top: 0;
  z-index: 3;
  width: 150px;
  border-right: 1px solid #fff;
  border-radius: 0;
  color: #fff;
  background-color: #0071bc;
  box-shadow: 3px 0 3px 1px rgba(0, 0, 0, 0.1);
  cursor: pointer; }
  .items_table .ui.sortable.table thead tr th:first-child {
    z-index: 4;
    box-shadow: 3px 0 3px 1px rgba(0, 0, 0, 0.1); }
  .items_table .ui.sortable.table thead tr th:hover, .items_table .ui.sortable.table thead tr th.sorted:hover {
    color: #fff;
    background: #205493; }
  .items_table .ui.sortable.table thead tr th label {
    color: #fff; }
