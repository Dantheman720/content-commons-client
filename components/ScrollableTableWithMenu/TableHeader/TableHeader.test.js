import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import TableHeader from './TableHeader';

const props = {
  tableHeaders: [
    { name: 'projectTitle', label: 'PROJECT TITLE' },
    { name: 'createdAt', label: 'CREATED' },
    { name: 'visibility', label: 'VISIBILITY' },
    { name: 'author', label: 'AUTHOR' }
  ],
  column: null,
  direction: null,
  handleSort: jest.fn(),
  displayActionsMenu: false
};

const Component = (
  <table>
    <TableHeader { ...props } />
  </table>
);

describe( '<TableHeader />', () => {
  it( 'renders without crashing', () => {
    const wrapper = mount( Component );
    const header = wrapper.find( 'TableHeader' );

    expect( header.exists() ).toEqual( true );
    expect( toJSON( header ) ).toMatchSnapshot();
  } );

  it( 'renders the correct table header values', () => {
    const wrapper = mount( Component );
    const th = wrapper.find( 'th' );

    th.forEach( ( t, i ) => {
      expect( t.text() ).toEqual( props.tableHeaders[i].label );
    } );
  } );

  it( 'renders the correct table header class value if displayActionsMenu', () => {
    const newProps = { ...props, ...{ displayActionsMenu: true } };
    const wrapper = mount(
      <table>
        <TableHeader { ...newProps } />
      </table>
    );
    const th = wrapper.find( 'th' );

    th.forEach( t => {
      expect( t.prop( 'className' ) ).toEqual( 'displayActionsMenu' );
    } );
  } );

  it( 'renders the correct table header sorted and direction values if `column === header.name`', () => {
    const newProps = {
      ...props,
      ...{ column: 'createdAt', direction: 'ascending' }
    };
    const wrapper = mount(
      <table>
        <TableHeader { ...newProps } />
      </table>
    );

    const cell = wrapper.find( 'TableHeaderCell' );

    cell.forEach( c => {
      if ( c.prop( 'children' ) === 'CREATED' ) {
        expect( c.prop( 'sorted' ) ).toEqual( newProps.direction );
      } else {
        expect( c.prop( 'sorted' ) ).toEqual( null );
      }
    } );
  } );
} );
