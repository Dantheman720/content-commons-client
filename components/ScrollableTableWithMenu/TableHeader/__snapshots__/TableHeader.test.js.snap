// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableHeader /> renders without crashing 1`] = `
Array [
  <TableHeader
    column={null}
    direction={null}
    displayActionsMenu={false}
    handleSort={
      [MockFunction] {
        "calls": Array [
          Array [
            "projectTitle",
          ],
          Array [
            "createdAt",
          ],
          Array [
            "visibility",
          ],
          Array [
            "author",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    tableHeaders={
      Array [
        Object {
          "label": "PROJECT TITLE",
          "name": "projectTitle",
        },
        Object {
          "label": "CREATED",
          "name": "createdAt",
        },
        Object {
          "label": "VISIBILITY",
          "name": "visibility",
        },
        Object {
          "label": "AUTHOR",
          "name": "author",
        },
      ]
    }
  >
    <TableHeader
      as="thead"
    >
      <thead
        className=""
      >
        <TableRow
          as="tr"
          cellAs="td"
        >
          <tr
            className=""
          >
            <TableHeaderCell
              as="th"
              className=""
              key="projectTitle"
              sorted={null}
            >
              <TableCell
                as="th"
                className=""
              >
                <th
                  className=""
                >
                  PROJECT TITLE
                </th>
              </TableCell>
            </TableHeaderCell>
            <TableHeaderCell
              as="th"
              className=""
              key="createdAt"
              sorted={null}
            >
              <TableCell
                as="th"
                className=""
              >
                <th
                  className=""
                >
                  CREATED
                </th>
              </TableCell>
            </TableHeaderCell>
            <TableHeaderCell
              as="th"
              className=""
              key="visibility"
              sorted={null}
            >
              <TableCell
                as="th"
                className=""
              >
                <th
                  className=""
                >
                  VISIBILITY
                </th>
              </TableCell>
            </TableHeaderCell>
            <TableHeaderCell
              as="th"
              className=""
              key="author"
              sorted={null}
            >
              <TableCell
                as="th"
                className=""
              >
                <th
                  className=""
                >
                  AUTHOR
                </th>
              </TableCell>
            </TableHeaderCell>
          </tr>
        </TableRow>
      </thead>
    </TableHeader>
  </TableHeader>,
  <TableHeader
    as="thead"
  >
    <thead
      className=""
    >
      <TableRow
        as="tr"
        cellAs="td"
      >
        <tr
          className=""
        >
          <TableHeaderCell
            as="th"
            className=""
            key="projectTitle"
            sorted={null}
          >
            <TableCell
              as="th"
              className=""
            >
              <th
                className=""
              >
                PROJECT TITLE
              </th>
            </TableCell>
          </TableHeaderCell>
          <TableHeaderCell
            as="th"
            className=""
            key="createdAt"
            sorted={null}
          >
            <TableCell
              as="th"
              className=""
            >
              <th
                className=""
              >
                CREATED
              </th>
            </TableCell>
          </TableHeaderCell>
          <TableHeaderCell
            as="th"
            className=""
            key="visibility"
            sorted={null}
          >
            <TableCell
              as="th"
              className=""
            >
              <th
                className=""
              >
                VISIBILITY
              </th>
            </TableCell>
          </TableHeaderCell>
          <TableHeaderCell
            as="th"
            className=""
            key="author"
            sorted={null}
          >
            <TableCell
              as="th"
              className=""
            >
              <th
                className=""
              >
                AUTHOR
              </th>
            </TableCell>
          </TableHeaderCell>
        </tr>
      </TableRow>
    </thead>
  </TableHeader>,
]
`;
